From a93fc67c0aba47bdee598e008ad2811bf10c4f95 Mon Sep 17 00:00:00 2001
Message-Id: <a93fc67c0aba47bdee598e008ad2811bf10c4f95.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Fri, 26 Sep 2014 10:14:52 +0530
Subject: [PATCH 302/312] FOR_UPSTREAM [VPG]: drm/i915: Fix in the MMIO flip
 for VLV

Patch fixes the MMIO flip function for VLV in terms of
calling the primary plane updates appropriately. This was
the root cause of all the z-order/alpha issues

Issue: GMINL-1364
Change-Id: I7b9b8a11caa558a1add9dd65e80220ef7f2bfc79
Signed-off-by: Pallavi G <pallavi.g@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |   19 ++++---------------
 1 file changed, 4 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 360b820..8ed127a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -9892,25 +9892,14 @@ static void intel_do_mmio_flip(struct intel_crtc *intel_crtc)
 	struct intel_framebuffer *intel_fb =
 		to_intel_framebuffer(intel_crtc->base.primary->fb);
 	struct drm_i915_gem_object *obj = intel_fb->obj;
+	struct drm_crtc *crtc =
+		dev_priv->pipe_to_crtc_mapping[intel_crtc->pipe];
 	u32 dspcntr;
 	u32 reg;
 
 	intel_mark_page_flip_active(intel_crtc);
-
-	reg = DSPCNTR(intel_crtc->plane);
-	dspcntr = I915_READ(reg);
-
-	if (INTEL_INFO(dev)->gen >= 4) {
-		if (obj->tiling_mode != I915_TILING_NONE)
-			dspcntr |= DISPPLANE_TILED;
-		else
-			dspcntr &= ~DISPPLANE_TILED;
-	}
-	I915_WRITE(reg, dspcntr);
-
-	I915_WRITE(DSPSURF(intel_crtc->plane),
-		   intel_crtc->unpin_work->gtt_offset);
-	POSTING_READ(DSPSURF(intel_crtc->plane));
+	dev_priv->display.update_primary_plane(crtc,
+		crtc->primary->fb, 0, 0);
 }
 
 static int intel_postpone_flip(struct drm_i915_gem_object *obj)
-- 
1.7.9.5

